runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -8362,8 +8362,28 @@ class INTERPRETER_CLASS
 			/* Restore SP to before popping memberNameObject. */
 			_sp -= 1;
 
-			/* Shift arguments by 1 and place memberNameObject before the first argument. */
-			memmove(_sp, _sp + 1, methodArgCount * sizeof(UDATA));
+			UDATA stackOffset = 1;
+			IDATA jitArgCount = (IDATA)_currentThread->floatTemp1;
+			/* Skip the below stack adjustment if JIT sets floatTemp1 to a negative value (-1). */
+			if (jitArgCount > 0) {
```

Comment:
`jitArgCount` == 0 would mean the ROM method had 0 args, but this would still push the appendix object which may be NULL.

Reply:
A `MemberName` is always expected here. So, `floatTemp1 (jitArgCount)` should never be 0 in `linkToStaticSpecial`.

Also, the `appendix object` will only exist for unresolved `invokedynamic` and `invokehandle` sequences. 

For skipping `stack adjustment`, the following condition will be true:
```
jitArgCount > 0 && methodArgCount > 0 && jitArgCount > methodArgCount
```